polars/polars-time/src/utils.rs

```diff
@@ -12,15 +12,24 @@ use polars_arrow::time_zone::Tz;
 use polars_core::prelude::{polars_bail, PolarsResult, TimeUnit};
 
 #[cfg(feature = "timezones")]
-pub(crate) fn localize_datetime(ndt: NaiveDateTime, tz: &Tz) -> PolarsResult<NaiveDateTime> {
+pub(crate) fn localize_datetime(
+    ndt: NaiveDateTime,
+    tz: &Tz,
+    use_earliest: Option<bool>,
+) -> PolarsResult<NaiveDateTime> {
     // e.g. '2021-01-01 03:00' -> '2021-01-01 03:00CDT'
     match tz.from_local_datetime(&ndt) {
         LocalResult::Single(tz) => Ok(tz.naive_utc()),
-        LocalResult::Ambiguous(_, _) => {
-            polars_bail!(
-                ComputeError: format!("datetime '{}' is ambiguous in time zone '{}'. Ambiguous datetimes are not yet supported", ndt, tz)
-            )
-        }
+        LocalResult::Ambiguous(dt_earliest, dt_latest) => match use_earliest {
+            Some(true) => Ok(dt_earliest.naive_utc()),
+            Some(false) => Ok(dt_latest.naive_utc()),
+            None => polars_bail!(ComputeError:
```

Comment:
Do we want to hit this branch? (and let the user make an active decision)? 

If so, I think `Option<bool>` is a valid type choice. If we are going for a default value `True | False` on the python side, I think we should just have a `bool` type.

Up to you! :)

Reply:
I think loudly erroring and forcing the user to choose, rather than picking for them, might be a better experience in the long-run - or at least, would be in-line with what polars usually does

It's not clear to me that one of the other would be the more correct one in general so I wouldn't know which to set as the default

Reply:
Ok, then I think this looks good to go! :)